File: r/difference-of-squares/difference-of-squares.R
# this is a stub function that takes a natural_number
# and should return the difference-of-squares as described
# in the README.md
square_of_sum <- function(number) {
  sum <-  number * (number + 1) / 2
  square <- sum ^ 2

  return(square)
}

sum_of_squares <- function(number) {
  return(number * (number + 1) * (2 * number + 1) / 6)
}

difference_of_squares <- function(number) {
  return(square_of_sum(number) - sum_of_squares(number))
}

File: python/difference-of-squares/difference_of_squares.py
"""Python Difference of Squares Exercism"""


def square_of_sum(number: int) -> int:
    """Calculates the square of sum.

    :param number: int
    :return: int
    """

    return (number * (number + 1) // 2) ** 2


def sum_of_squares(number: int) -> int:
    """Calculates the sum of squares.

    :param number: int
    :return: int
    """

    return (number * (number + 1) * (2 * number + 1)) // 6


def difference_of_squares(number: int) -> int:
    """Calculates the difference of squares.

    :param number: int
    :return: int
    """

    return square_of_sum(number) - sum_of_squares(number)
